### DISCUSSION TITLE
Saving Current Layout State

### DISCUSSION THREAD

**Discussion by kuket15** [KevinVandy/material-react-table#350] [Q&A]
Hello and thanks for such a good library.

I wonder if there is a builtin mechanism to save the table layout after user has changed columns order, columns width and so on...

I'd like to persist it in redux or maybe server side and load it when the user reloads the page.

Thanks is advance

Luca

**Comment by KevinVandy** [comment]
You can manage whichever state you need to yourself. https://www.material-react-table.com/docs/guides/table-state-management#manage-individual-states-as-needed

Once you've hoisted up the state you want to manage, you can do whatever you want with it. Like persist it in local storage, redux, a database, or whatever.
